f5_openstack_agent/lbaasv2/drivers/bigip/icontrol_driver.py

```diff
@@ -1381,19 +1399,25 @@ def _update_listener_status(self, service):
                         lb_const.OFFLINE)
 
     @log_helpers.log_method_call
-    def _update_loadbalancer_status(self, service):
+    def _update_loadbalancer_status(self, service, timed_out=False):
         """Update loadbalancer status in OpenStack """
-        loadbalancer = service['loadbalancer']
-        provisioning_status = loadbalancer['provisioning_status']
+        loadbalancer = service.get('loadbalancer', {})
+        provisioning_status = loadbalancer.get('provisioning_status',
+                                               plugin_const.ERROR)
 
         if (provisioning_status == plugin_const.PENDING_CREATE or
                 provisioning_status == plugin_const.PENDING_UPDATE):
-            operating_status = (lb_const.ONLINE)
+            if timed_out:
+                operating_status = (lb_const.OFFLINE)
```

Comment:
If a member is timed_out, you set prov_status to ERROR.  Why not do the same here?  I think we've failed to fully provision the loadbalancer due to missing network configuration.